enclave-runtime/src/teeracle/mod.rs

```diff
@@ -41,6 +44,104 @@ use sgx_types::sgx_status_t;
 use sp_runtime::OpaqueExtrinsic;
 use std::{string::String, vec::Vec};
 
+fn update_weather_data_internal(weather_info: WeatherInfo) -> Result<Vec<OpaqueExtrinsic>> {
+	let extrinsics_factory = GLOBAL_EXTRINSICS_FACTORY_COMPONENT.get()?;
+	let ocall_api = GLOBAL_OCALL_API_COMPONENT.get()?;
+
+	let mut extrinsic_calls: Vec<OpaqueCall> = Vec::new();
+
+	let open_meteo_weather_oracle = create_open_meteo_weather_oracle(ocall_api);
+
+	match get_longitude(weather_info, open_meteo_weather_oracle) {
+		Ok(opaque_call) => extrinsic_calls.push(opaque_call),
+		Err(e) => {
+			error!("[-] Failed to get the newest longitude from OpenMeteo. {:?}", e);
+		},
+	};
+	let extrinsics = extrinsics_factory.create_extrinsics(extrinsic_calls.as_slice(), None)?;
+	Ok(extrinsics)
+}
+
+fn get_longitude<OracleSourceType, MetricsExporter>(
+	weather_info: WeatherInfo,
+	oracle: WeatherOracle<OracleSourceType, MetricsExporter>,
+) -> Result<OpaqueCall>
+where
+	OracleSourceType: OracleSource<
+		WeatherInfo,
+		OracleRequestResult = std::result::Result<f32, ita_oracle::error::Error>,
+	>,
+	MetricsExporter: ExportMetrics<WeatherInfo>,
+{
+	let longitude =
+		oracle.get_longitude(weather_info.clone()).map_err(|e| Error::Other(e.into()))?;
+
+	let base_url = oracle.get_base_url().map_err(|e| Error::Other(e.into()))?;
+	let source_base_url = base_url.as_str();
+
+	println!("Update the longitude:  {}, for source {}", longitude, source_base_url);
+
+	let node_metadata_repository = GLOBAL_NODE_METADATA_REPOSITORY_COMPONENT.get()?;
+
+	let call_ids = node_metadata_repository
+		.get_from_metadata(|m| m.update_oracle_call_indexes())
+		.map_err(Error::NodeMetadataProvider)?
+		.map_err(|e| Error::Other(format!("{:?}", e).into()))?;
+
+	let call = OpaqueCall::from_tuple(&(
+		call_ids,
+		weather_info.weather_query.key().as_bytes().to_vec(),
+		source_base_url.as_bytes().to_vec(),
+		// TODO: Does this work? the type the extrinsic is expecting is BoundedVec<u8, 4096>
+		Some(longitude.encode()),
+	));
+
+	Ok(call)
+}
+
+// TODO Question: What can be fed to this function? Can a struct be passed? if so how?
```

Comment:
You can pass in any Rust type, as long as it implements `Encode`/`Decode`. You can have a look at some of the other E-calls/O-calls, where we pass in encoded structs or collections.